<!doctype html public "-//w3c//dtd html 4.0 transitional//en">
<html>
<head>
   <meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
   <meta name="GENERATOR" content="Mozilla/4.75 [en] (Win98; U) [Netscape]">
   <title>Correspondence Analysis of Codon Usage</title>
</head>
<body bgcolor="#FFFFFF" link="#0000FF" vlink="#FF00FF" background="edge.gif" leftmargin="110">
&nbsp;
<table BORDER=0 WIDTH="620" >
<tr>
<td VALIGN=TOP WIDTH="95"></td>

<td VALIGN=TOP WIDTH="515">
<center><b><font face="Helvetica,Arial"><font size=+2>Correspondence Analysis
of Codon Usage</font></font></b>
<p><img SRC="bar.gif" height=15 width=439></center>

<p><font size=+0><a href="ftp://molbiol.ox.ac.uk/cu/codonW.tar.Z">CodonW</a>
is a programme designed to simplify the Multivariate analysis (correspondence
analysis) of codon and amino acid usage. It also calculates standard indices
of codon usge. It has both menu and command-line interfaces. It was written
by <a href="mailto:johnp@molbiol.ox.ac.uk">John Peden</a> as part of his
PhD project in the lab of
<a href="http://evol.nott.ac.uk/pms.html">Paul
Sharp, </a><a href="http://evol.nott.ac.uk/cmelun/index.html">CMELUN,</a>
<a href="http://evol.nott.ac.uk/research.html">Dept of Genetics,</a> <a href="http://www.nott.ac.uk/">University
of Nottingham</a>. John is currently employed at the
<a href="http://www.mrc.ac.uk">MRC
</a>Haematology Unit at the <a href="http://www.imm.ox.ac.uk">IMM </a>who
are kindly hosting this page.</font>
<p><b><font size=+0>NB: I have extensively tested it and believe it to
be bug free. If you find a bug or something confusing please email me at
<a href="mailto:john.peden@worf.molbiol.ox.ac.uk">John.Peden@molbiol.ox.ac.uk</a>
and let me know </font></b><font size=+0>(Don't forget to tell me the Operating
System and to send me the input file that generated the error).</font>
<ul>
<li>
<i><font color="#008040"><a href="#Downloading and Installation">Downloading
and Installation</a></font></i></li>

<li>
<i><font color="#008040"><a href="#Features:">Features</a></font></i></li>

<li>
<a NAME="System and Hardware" href="http://www.molbiol.ox.ac.uk/cu/culong.html#System and Hardware"></a><i><font color="#008040"><a href="http://www.molbiol.ox.ac.uk/cu/culong.html#System and Hardware" name="System and Hardware">System
and Hardware&nbsp;</a></font></i></li>

<li>
<a NAME="CodonW" href="http://www.molbiol.ox.ac.uk/cu/culong.html#CodonW"></a><i><font color="#008040"><a href="http://www.molbiol.ox.ac.uk/cu/culong.html#CodonW" name="CodonW">CodonW</a></font></i></li>

<li>
<a NAME="Introduction to Codon Usage" href="http://www.molbiol.ox.ac.uk/cu/culong.html#Introduction to Codon Usage"></a><i><font color="#008040"><a href="http://www.molbiol.ox.ac.uk/cu/culong.html#Introduction to Codon Usage" name="Introduction to Codon Usage">Introduction
to Codon Usage&nbsp;</a></font></i></li>

<li>
<a NAME="Correspondence Analysis" href="http://www.molbiol.ox.ac.uk/cu/culong.html#Correspondence Analysis"></a><i><font color="#008040"><a href="http://www.molbiol.ox.ac.uk/cu/culong.html#Correspondence Analysis" name="Correspondence Analysis">Correspondence
Analysis</a></font></i></li>

<li>
<a NAME="Input files" href="http://www.molbiol.ox.ac.uk/cu/culong.html#Input files"></a><i><font color="#008040"><a href="http://www.molbiol.ox.ac.uk/cu/culong.html#Input files" name="Input files">Input
files</a></font></i></li>

<li>
<a NAME="The Menu Interface" href="http://www.molbiol.ox.ac.uk/cu/culong.html#The Menu Interface"></a><i><font color="#008040"><a href="http://www.molbiol.ox.ac.uk/cu/culong.html#The Menu Interface" name="The Menu Interface">The
Menu Interface</a></font></i></li>

<li>
<a NAME="Correspondence analysis in CodonW." href="http://www.molbiol.ox.ac.uk/cu/culong.html#Correspondence analysis in CodonW."></a><i><font color="#008040"><a href="http://www.molbiol.ox.ac.uk/cu/culong.html#Correspondence analysis in CodonW." name="Correspondence analysis in CodonW.">Correspondence
analysis in CodonW</a></font></i></li>

<li>
<i><font color="#008040"><a href="http://www.molbiol.ox.ac.uk/cu/culong.html#License">License</a></font></i></li>

<li>
<a NAME="References" href="http://www.molbiol.ox.ac.uk/cu/culong.html#References"></a><i><font color="#008040"><a href="http://www.molbiol.ox.ac.uk/cu/culong.html#References" name="References">References</a></font></i></li>

<li>
<i><font color="#008040"><a href="http://www.molbiol.ox.ac.uk/cu/culong.html#Name">Contact</a></font></i></li>
</ul>

<center><img SRC="bar.gif" height=15 width=439></center>

<p><a NAME="Downloading and Installation"></a><font face="Helvetica,Arial"><font color="#804000">Downloading
and Installation</font></font>
<p>Binaries are available for the following operating systems. Alternatively
you can download the source code in <a href="ftp://molbiol.ox.ac.uk/cu/codonW.tar">tar</a>,
<a href="ftp://molbiol.ox.ac.uk/cu/codonW.tar.Z">compressed</a> or
<a href="ftp://molbiol.ox.ac.uk/cu/codonW.tar.gz">gzipped
</a>format and compile it yourself. See the
<a href="http://molbiol.ox.ac.uk/cu/Readme.html">README</a>
file for more information.
<p>1. <a href="ftp://worf.molbiol.ox.ac.uk/cu/DU4codonW.tar.Z">Digital
UNIX 4.0</a>
<p>2. <a href="ftp://molbiol.ox.ac.uk/cu/Irix64codonW.tar.Z">Irix 6.5</a>
(64 bit)
<p>3. <a href="ftp://molbiol.ox.ac.uk/cu/SolarisCodonW.tar.Z">Solaris 2.6
</a>(solaris)
<p>4. <a href="ftp://molbiol.ox.ac.uk/cu/Win32codonW.zip">Win 32bit (NT/95/98/2000)</a>
(No GUI; runs within a DOS window)
<p>5.(Sorry no Mac version the source code should compile on CodeWarrior
but I don't have a MAC compiler, bewarned it doesn't have any GUI)
<p>6. Web access via <a href="http://www-alt.pasteur.fr/~letondal/Pise/">PISE</a>
at the pasteur <a href="http://bioweb.pasteur.fr/seqanal/interfaces/codonw.html">http://bioweb.pasteur.fr/seqanal/interfaces/codonw.html</a>
<center>
<p><img SRC="bar.gif" height=15 width=439></center>

<p><a NAME="Features:"></a><font face="Helvetica,Arial"><font color="#804000">Features:</font></font>
<ol>
<li>
<b>Written in ANSI compliant C</b></li>

<li>
<b>Menu driven</b></li>

<li>
<b>Extensive number of command line options</b></li>

<li>
<b>Genetic code independent</b><font size=+1>&nbsp;</font></li>

<li>
<font size=+1><b>No limits on</b><i>&nbsp;</i></font></li>

<ol type="a" start="1">
<li>
<i><font size=+1>number of sequences&nbsp;</font></i></li>

<li>
<i><font size=+1>sequence length</font></i></li>
</ol>

<li>
<b><font size=+1>Calculates the codon usage indices</font></b></li>

<ol type="a"
        start="1">&nbsp;
<li>
<i><font size=+1>CAI: Codon adaptation index</font></i></li>

<li>
<i><font size=+1>Fop: Frequency of optimal codons</font></i></li>

<li>
<i><font size=+1>Nc: Effective number of codons</font></i></li>

<li>
<i><font size=+1>CBI: Codon Bias Index&nbsp;</font></i></li>
</ol>

<li>
<b><font size=+1>Calculates amino acid indices</font></b></li>

<ol 
        type="a">&nbsp;
<li>
<i><font size=+1>GRAVY score</font></i></li>

<li>
<i><font size=+1>Aromaticity</font></i></li>
</ol>

<li>
<b><font size=+1>Calculates correspondence analysis of&nbsp;</font></b></li>

<ol type="a">
<li>
<i><font size=+1>Codon Usage</font></i></li>

<li>
<i><font size=+1>RSCU (Relative synonymous codon usage)</font></i></li>

<li>
<i><font size=+1>Amino acid usage</font></i></li>
</ol>

<li>
<b><font size=+1>Correspondence analysis&nbsp;</font></b></li>

<ol type="a">
<li>
<i><font size=+1>can include/exclude codons/amino acids</font></i></li>

<li>
<i><font size=+1>can generate detailed reports of trends</font></i></li>

<li>
<i><font size=+1>attempts to identify optimal codons automatically</font></i></li>

<li>
<i><font size=+1>can allow additional data sets to be added&nbsp;</font></i></li>

<li>
<i><font size=+1>records any number of trends</font></i></li>
</ol>

<li>
<b><font size=+1>Calculates gene parameters</font></b></li>

<ol 
        type="a">&nbsp;
<li>
<i><font size=+1>Gene length</font></i></li>

<li>
<i><font size=+1>GC, GC3s and codon position specific G+C&nbsp;</font></i></li>

<li>
<i><font size=+1>Dinucleotide composition (in all three codon frames)</font></i></li>

<li>
<i><font size=+1>Amino acid usage</font></i></li>

<li>
<i><font size=+1>Relative amino acid usage</font></i></li>

<li>
<i><font size=+1>Codon usage</font></i></li>

<li>
<i><font size=+1>Relative Synonymous codon usage</font></i></li>
</ol>

<li>
<b><font size=+1>Can conceptually translates sequences to protein</font></b></li>

<li>
<b><font size=+1>Reformats sequence data</font></b></li>

<li>
<b><font size=+1>Human or machine readable output</font></b></li>

<li>
<b><font size=+1>Can concatenate genes (whilst preserving reading frame)
to calculate</font></b></li>

<ol 
        type="a">&nbsp;
<li>
<i><font size=+1>overall codon usage</font></i></li>

<li>
<i><font size=+1>GC content</font></i></li>

<li>
<i><font size=+1>Dinucleotide composition&nbsp;</font></i></li>
</ol>

<li>
<b><font size=+1>Generates tables of codon, amino acid, or RSCU usage</font></b></li>

<li>
<b><font size=+1>Can even help you learn the Genetic code.</font></b></li>

<li>
<b><font size=+1>and more</font></b></li>
</ol>

<center><img SRC="bar.gif" height=15 width=439></center>

<p><font size=+1>For more information about codonW and codon usage analysis
click <a href="http://www.molbiol.ox.ac.uk/cu/culong.html">here</a> or
read the <a href="http://www.molbiol.ox.ac.uk/cu/Readme.html">documentation</a>
that comes with codonw.&nbsp;</font>
<center>
<p><img SRC="bar.gif" height=15 width=439>
<p><img SRC="nph-count" height=20 width=60></center>

<p><font size=+1>Page maintained by&nbsp;<a NAME="Name" href="mailto:johnp@molbiol.ox.ac.uk"></a><a href="mailto:johnp@molbiol.ox.ac.uk" name="Name">John
Peden</a>, last updated 06/06/97.&nbsp;</font>
<p><font size=+1>Back to the OUBC<a href="http://www.molbiol.ox.ac.uk">
home page</a>.&nbsp;</font>
<p><font size=+1>Like the graphics?. They were kindly provided by <a href="http://www.erinet.com/jelane/families/">Jelane
Johnson</a>.&nbsp;</font>
<center>
<p><img SRC="bar.gif" height=15 width=439></center>
</td>
</tr>
</table>

</body>
</html>
